src/views/dialogs/NewMarketMessageDetailsDialog.tsx

```diff
@@ -0,0 +1,354 @@
+import { useMemo, useState } from 'react';
+import styled, { AnyStyledComponent } from 'styled-components';
+import { utils } from '@dydxprotocol/v4-client-js';
+
+import { STRING_KEYS } from '@/constants/localization';
+import { isMainnet } from '@/constants/networks';
+import { PotentialMarketItem } from '@/constants/potentialMarkets';
+import { useGovernanceVariables, useStringGetter, useTokenConfigs } from '@/hooks';
+import { usePotentialMarkets } from '@/hooks/usePotentialMarkets';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { Details } from '@/components/Details';
+import { Dialog } from '@/components/Dialog';
+import { Output, OutputType } from '@/components/Output';
+import { Tag, TagType } from '@/components/Tag';
+import { ToggleGroup } from '@/components/ToggleGroup';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+type ElementProps = {
+  preventClose?: boolean;
+  setIsOpen?: (open: boolean) => void;
+  assetData?: PotentialMarketItem;
+  clobPairId?: number;
+  liquidityTier?: string;
+};
+
+export enum CodeToggleGroup {
+  CREATE_ORACLE = 'CREATE_ORACLE',
+  MSG_CREATE_PERPETUAL = 'MSG_CREATE_PERPETUAL',
+  MSG_CREATE_CLOB_PAIR = 'MSG_CREATE_CLOB_PAIR',
+  MSG_DELAY_MESSAGE = 'MSG_DELAY_MESSAGE',
+  MSG_SUBMIT_PROPOSAL = 'MSG_SUBMIT_PROPOSAL',
+}
+
+export const NewMarketMessageDetailsDialog = ({
+  assetData,
+  clobPairId,
+  liquidityTier,
+  preventClose,
+  setIsOpen,
+}: ElementProps) => {
+  const [codeToggleGroup, setCodeToggleGroup] = useState(CodeToggleGroup.CREATE_ORACLE);
+  const { exchangeConfigs } = usePotentialMarkets();
+  const { baseAsset } = assetData ?? {};
+  const { newMarketProposal } = useGovernanceVariables();
+  const stringGetter = useStringGetter();
+  const { chainTokenLabel } = useTokenConfigs();
+  const initialDepositAmountDecimals = isMainnet ? 0 : 18;
+
+  const exchangeConfig = useMemo(() => {
+    return baseAsset ? exchangeConfigs?.[baseAsset] : undefined;
+  }, [baseAsset]);
+
+  const ticker = useMemo(() => `${baseAsset}-USD`, [baseAsset]);
+
+  const toggleGroupItems: Parameters<typeof ToggleGroup>[0]['items'] = useMemo(() => {
+    return [
+      {
+        value: CodeToggleGroup.CREATE_ORACLE,
+        label: 'Create oracle market',
```

Comment:
This is the name of one of the `Msgs` used for the proposal. I am intentionally not localizing anything pertaining to parameters so that it stays true to the actual interface.